backends/ebpf/ebpfProgram.cpp

```diff
@@ -198,6 +199,9 @@ void EBPFProgram::emitPreamble(CodeBuilder *builder) {
         "{ u8 mask = EBPF_MASK(u8, s); "
         "*((u8*)a) = ((*((u8*)a)) & ~mask) | (((v) >> (8 - (s))) & mask); "
         "} while (0)");
+    builder->appendLine(
+        "#define write_partial_ex(a, w, s, v) do { *((u8*)a) = ((*((u8*)a)) "
+        "& ~(EBPF_MASK(u8, w) << s)) | (v << s) ; } while (0)");
```

Comment:
Excluding definition of `memcpy` definitions here looks the same as for PSA arch. I think it would be better to move (or create) `emitCommonPreamble` to `EBPFPrgram` class with this common code. From `PSAEbpfGenerator` it could be called via `ingress` field.

Reply:
Thanks for the pointers - that is indeed a cleaner way to do things. Rewrote the PR accordingly.